BackupDatabase-Recipe.rb

```diff
@@ -0,0 +1,76 @@
+require 'nokogiri'
+
+def get_database_credentials_for_magento
+  configXMLFile = Nokogiri::XML(File.open("#{current_release}#{app_path}/app/etc/local.xml"))
+  dbCredentials = Hash.new
+  ["host", "username", "password", "dbname"].each do |credential|
+    dbCredentials[credential] = configXMLFile.xpath("//config//global//resources//default_setup//#{credential}/text()").text
+  end
+  dbCredentials
+end
+
+def get_database_credentials_for_cakephp
+  configFile = File.open("#{current_release}#{app_path}/Config/database.php").read
+
+  rowCredentials= configFile.scan(/'(?<key>\w+)'\s=>\s'(?<value>\w+)'/)
+  dbCredentials = Hash.new
+
+  mapping = {"login" => "username", "database" => "dbname"}
+  rowCredentials.each do |key,value|
+    credentialKey = mapping[key].nil? ? key : mapping[key]
+    dbCredentials[credentialKey] = value
+  end
+
+  dbCredentials
+end
+
+def is_cakephp_project
+  exists?(:cake_migrations)
+end
+
+namespace :database do
+
+  desc <<-DESC
+		Generate a backup for database project.
+  DESC
+  task :backup, :roles => :db, :only => { :primary => true } do
+    filename = "#{application}.dump.sql.gz"
+    file = "/tmp/#{filename}"
+    on_rollback { delete file }
+
+    logger.info("Fetching database credentials from project")
+    dbCredentials = is_cakephp_project ? get_database_credentials_for_cakephp : get_database_credentials_for_magento
```

Comment:
Agreed with @real34.

A (maybe) even better solution should to use the capacity of Capistrano to let the user declare servers and roles like so:

```ruby
server 'my-db.example.com', :db, {
  :primary => true
  :user => 'user',
  :password => 'password',
}
# a method like Cake::dbCredentials() could be used to get those too
```

then:
```ruby
task :backup, :roles => :db, :only => { :primary => true } do
  find_servers_for_task(current_task).each do |server|
    # […]
  end
end
```

I did not search any further, but `server` should be an instance of [ServerDefinition](http://ruby-doc.org/gems/docs/c/capistrano-edge-2.5.6/Capistrano/ServerDefinition.html) and the `user`/`password` data should be in the `options` attribute if I read the docs well.

The good thing with this solution is that it uses the full potential of Capistrano when it comes to have several databases for the same stage (dev, prod, etc.) on several servers.